src/llama-context.cpp

```diff
@@ -449,7 +447,7 @@ uint32_t llama_context::n_ctx() const {
 }
 
 uint32_t llama_context::n_ctx_per_seq() const {
-    return cparams.n_ctx / cparams.n_seq_max;
+    return cparams.kv_unified ? cparams.n_ctx : cparams.n_ctx / cparams.n_seq_max;
```

Comment:
Should this value be capped when using unified cache to avoid exceeding the model context length? I think it could be set to `min(n_ctx_train, n_ctx)`, or add a parameter to allow the user to change it.